src/llama.cpp

```diff
@@ -22057,6 +22085,14 @@ static int32_t llama_chat_apply_template_internal(
         if (add_ass) {
             ss << "<|assistant|>";
         }
+    } else if(tmpl == "glm-edge" || tmpl_contains("<|assistant|>")){
```

Comment:
For future-proof, I would recommend making sure the template does not contain `<|end|>` or `</s>` token:

```suggestion
    } else if(tmpl == "glm-edge" || (tmpl_contains("<|assistant|>") && !tmpl_contains("<|end|>") && !tmpl_contains("</s>"))){
```

Although, if you are from THUDM team, I would recommend to have EOS token at the end of each turn in your future models. This will simplify downstream implementation if all models follow this rule.